airflow/models/__init__.py

```diff
@@ -1730,6 +1730,8 @@ def email_alert(self, exception):
         # For reporting purposes, we report based on 1-indexed,
         # not 0-indexed lists (i.e. Try 1 instead of
         # Try 0 for the first attempt).
+        # This function is called after changing the state
+        # from State.RUNNING so need to subtract 1 from self.try_number.
```

Comment:
Could you move this comment to line 1720 instead of here?

Reply:
No worries. I've moved it.

Reply:
@TheZepto I am not very familiar with this part of the code. But based on this line of comment(https://github.com/apache/airflow/blob/master/airflow/models/__init__.py#L1523), it seems the try_number hasn't increased yet. Do you happen to know which line increases the try number causing the issue?

Reply:
My understanding is that this comment "Since this function is called only when the TI state is running, try_number contains the current try_number (not the next)." is saying that returning try_number with the state as running returns the current try_number. However, changing the state causes try_number to return as the next try_number (https://github.com/apache/airflow/blob/master/airflow/models/__init__.py#L706).

I'm not sure why the behaviour of try_number is to be incremented whenever the state is not equal to running.

Reply:
thanks.